Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -3827,7 +3855,41 @@ class LowerToAir {
             }
             
             if constexpr (isX86()) {
-                append(VectorSplatInt8, shiftAmount, shiftVector);
+                append(Move, shift, shiftAmount);
+                append(And32, Arg::imm(mask), shiftAmount);
+                if (value->opcode() == VectorShr && value->signMode() == SIMDSignMode::Signed && value->simdLane() == SIMDLane::i64x2) {
+                    // x86 has no SIMD 64-bit signed right shift instruction, so we scalarize it here.
+
+#if CPU(X86_64)
```

Comment:
This is already in an if?

Reply:
The first `if` doesn't protect from the fact we use `X86Registers::ecx`, which won't compile on non-Intel. So I guess we could make the whole thing part of the `#if CPU(X86_64)`.  Better would just be to get rid of it though, since this class also defines `m_ecx` which I could just use instead. I'll use that instead and remove the preprocessor directives.